rpcserver.go

```diff
@@ -2711,32 +2711,66 @@ func (r *rpcServer) WalletBalance(ctx context.Context,
 // ChannelBalance returns the total available channel flow across all open
 // channels in satoshis.
 func (r *rpcServer) ChannelBalance(ctx context.Context,
-	in *lnrpc.ChannelBalanceRequest) (*lnrpc.ChannelBalanceResponse, error) {
+	in *lnrpc.ChannelBalanceRequest) (
+	*lnrpc.ChannelBalanceResponse, error) {
+
+	var (
+		localBalance             btcutil.Amount
+		remoteBalance            btcutil.Amount
+		unsettledBalance         btcutil.Amount
+		pendingOpenLocalBalance  btcutil.Amount
+		pendingOpenRemoteBalance btcutil.Amount
+
+		// Deprecated fields
+		balance            btcutil.Amount
+		pendingOpenBalance btcutil.Amount
+	)
 
 	openChannels, err := r.server.chanDB.FetchAllOpenChannels()
 	if err != nil {
 		return nil, err
 	}
 
-	var balance btcutil.Amount
 	for _, channel := range openChannels {
-		balance += channel.LocalCommitment.LocalBalance.ToSatoshis()
+		c := channel.LocalCommitment
+		localBalance += c.LocalBalance.ToSatoshis()
+		remoteBalance += c.RemoteBalance.ToSatoshis()
+
+		balance += c.LocalBalance.ToSatoshis()
```

Comment:
Since `balance` is the same as `localBalance`, you could move this out of the loop.

Reply:
fixed